tests/Doctrine/Tests/Common/Annotations/AbstractReaderTest.php

```diff
@@ -501,38 +602,56 @@ class TestIgnoresNonAnnotationsClass
 
 class TestTopLevelAnnotationClass
 {
-    /** @\TopLevelAnnotation */
-    private $field;
+    /**
+     * @var mixed
```

Comment:
Aren't these `@var` annotations testing the ignored annotations again? In case someone breaks the ignored-annotations functionality, wouldn't all the additions you've added break more tests in such a case than necessary?

Reply:
I guess they are, and would, but you have to choose between that and ending up with a lot more ignore rules in the phpcs config files, so I figured we would be better off that way than potentially ignoring a lot more violations that we mean to. Getting properly-annotated properties in actual test classes (as opposed to fixtures) helps static analysis a lot, and can avoid a lot of mistakes.

Reply:
This is a "pick your poison" situation. Considering that we won't check future PRs which will introduce new classes with the need to extend the phpcs ignore rules, the lesser evil would be keeping the annotations.

Reply:
> This is a "pick your poison" situation.

:100: Wouldn't have phrased it better!